### ISSUE TITLE
Requires g++ 4.6

### ISSUE THREAD

**Issue by davidljung** [RcppCore/RcppArmadillo#97]
The latest version requires g++ 4.6 in the configure script.  I'm wondering if this is due to the use of some newer C++ features in the source, or if it doesn't really need 4.6.

We run R on an IBM Netezza PureData appliance box, which is a managed R environment.  Hence, we don't have the ability to upgrade g++. RcppArmadillo is a dependency of an R package we use, but we're unable to install version 0.7.100.3.0 due to this requirement.

```
configure: WARNING: Only g++ version 4.6 or greater can be used with RcppArmadillo.
configure: error: Please use a different compiler.
ERROR: configuration failed for package RcppArmadillo
```

Consequently, we're explicitly installing the CRAN archive version 0.6.700.6.0, which achieves what we need.

If this is a real requirement, please just close this request, but if not we'd like to suggest lowering the g++ version requirement in the configuration to the previous version that was specified.

Presumably, IBM will update the NZR software in the appliance in future, but in the enterprise environment that typically takes considerable time.

Thanks!

**Comment by eddelbuettel** [comment]
Why do you open an issue?  

You need to install an older version of RcppArmadillo that is approrpriately dated for your compiler.   And you did.  So what more is there to say, really?

**Comment by kevinushey** [comment]
Indeed, the latest versions of `armadillo` + `RcppArmadillo` require a newer compiler, hence the `configure` warnings.

**Comment by eddelbuettel** [comment]
Error, actually -- see [here](https://github.com/RcppCore/RcppArmadillo/blob/master/configure.ac#L31).

No point having a warning only to have the compilation die a minute later. This is a *hard* requirement (from upstream).

**Comment by jrounds** [comment]
Any advice on which version to install?
https://cran.r-project.org/src/contrib/Archive/RcppArmadillo/

I am not quite sure on an efficient way to discover an old version compatible with a specific compiler.

**Comment by eddelbuettel** [comment]
To a first approximation follow the Armadillo majors, so try the last 6.* release: 0.6.700.6.0.  Then 0.5.600.2.0 and so on.

**Comment by jrounds** [comment]
Thanks.  Picked Dec because holidays!  Seems sufficient (wanted latest
Forecast package).  Sorry for the hassle I should have tried a few before
posting.

**Comment by pengchy** [comment]
Hi, sorry to comment on this issue, but I can solve my question.

I have successfully installed latest Rcpp version 0.12.8 and RcppArmadillo version 0.6.100.0.0 with g++ 4.4.6. But one of my wanted package "NetDiff" (https://gitlab.com/tt104/NetDiff) need the latest RcppArmadillo version 0.7.600.1.0. However, when I add the g++ version 4.9.2, I got the following  error message:

```
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/tmp/Rtmp5LKNdB/R.INSTALL6ffa4a2cc4a4/RcppArmadillo':
configure: error: cannot run C++ compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
ERROR: configuration failed for package 'RcppArmadillo'
```

Is it possible to compile RcppArmadillo_0.7.600.1.0 with g++ 4.9.2 when I have compiled other package with g++ 4.4.6?

Thanks a lot and Happy New Year!
Pengcheng Yang

**Comment by eddelbuettel** [comment]
You may need to alk to the NetDiff maintainers.  Maybe you can just edit out the requirement from their DESCRIPTION. Not our package.

But a current (Rcpp)Armadillo *requires* a new-enough compiler. And g++ 4.4.6 on ancient RHEL or CentOs does not work.  Sorry.

**Comment by pengchy** [comment]
Hi Eddelbuettel,

Thank you for your reply. 

I have noticed that the error message says: "If you meant to cross compile, use `--host'." So, I wander if it possible compile RcppArmadillo_0.7.600.1.0 using "cross compile with --host". I have uncompressed the package, but don't know how to do "cross compile use --host".

Anyway, I will try to edit the DESCRIPTION file of NetDiff to use the older version of RcppArmadillo.

Thanks!

**Comment by eddelbuettel** [comment]
[ The first name is 'Dirk' ]

That message does not come from us; I suspect you may just have used `configure` the wrong way.

**Comment by pengchy** [comment]
Hi Dirk,

The installation was performed using the command "R CMD INSTALL", so I don't know how to `configure`, such as add "--host" parameter, when installing the R package. OK, I will searched how to do this work.

I have contacted with the NetDiff maintainer. Thank you!

**Comment by majaidi** [comment]
I wanted to let everyone that I have installed version 0.5 RcppArmadillo to compile the forecast package successfully. No need to get a new GCC :)

**Comment by ppadebettu** [comment]
Hello majaidi, what are the steps you followed to install version 0.5 of RcppArmadillo without updating GCC ?

**Comment by eddelbuettel** [comment]
Go to the [archive page for RcppArmadillo](https://cran.r-project.org/src/contrib/Archive/RcppArmadillo/) and download the desired version.  You can use the archive file stamp along with the [NEWS](https://cloud.r-project.org/web/packages/RcppArmadillo/news.html) file to guide.   Download the tarball, install locally from source as usual.

**Comment by antass** [comment]
This is old, but FWIW, I was able to figure out which version of RcppArmadillo to get by checking [which version of ubuntu was shipped with my version of g++ (v4.6)](https://packages.ubuntu.com/trusty/g++-4.6). Trusty came out in April 2014, so I downloaded a version of RcppArmadillo from early 2014 (v0.4). It worked.

> Any advice on which version to install?
> https://cran.r-project.org/src/contrib/Archive/RcppArmadillo/
> 
> I am not quite sure on an efficient way to discover an old version compatible with a specific compiler.

**Comment by eddelbuettel** [comment]
Even easier: install the `r-cran-rcpparmadillo` that _came with your version of Ubuntu_ if you prefer to (or are forced to) run outdated code.
